django/thunderstore/repository/views/repository.py

```diff
@@ -246,7 +246,7 @@ def get_queryset(self):
                 review_status=PackageListingReviewStatus.rejected,
             )
 
-        search_query = self.get_search_query()
+        search_query = self.get_search_query().rstrip(" ")
```

Comment:
This isn't enough to solve the issue. For one, a leading space will still have the bug, but even if you used `strip` instead of `rstrip`, double spaces in the middle would still have the same problem.

I'd recommend applying the fix directly on `perform_search`, by replacing
```python
parts = search_query.split(" ")
```
with
```python
parts = [x for x in search_query.split(" ") if x]
```